src/Elastic.Apm/Metrics/MetricsCollector.cs

```diff
@@ -93,8 +93,30 @@ internal void CollectAllMetricsImpl()
 		{
 			_logger.Trace()?.Log("CollectAllMetrics started");
 
-			var samplesFromAllProviders = new List<MetricSample>();
+			var samplesFromAllProviders = CollectMetricsFromProviders();
```

Comment:
If we add here

```
if (samplesFromAllProviders.IsEmpty())
{
    _logger.Debug()?.Log("No metrics collected (no provider returned valid samples) - nothing to send to APM Server");
    return;
}
```
It will achieve the requirement of not sending empty metric sets. IMHO making payload sender filter out empty metric sets breaks encapsulation - all metrics related logic should stay in metrics related code.
I think this requirement is related to this PR so there is no need for a separate PR.

Of course, `CollectAllMetrics_ShouldDisableProvider_WhenSamplesAreInvalid` test will need to be updated:

```
mockPayloadSender.Metrics.Should().BeEmpty();
```

instead of

```
mockPayloadSender.Metrics.Count.Should().Be(iterations);
mockPayloadSender.Metrics.Should().OnlyContain(x => !x.Samples.Any());
```

Reply:
done